controller/clusters_blackbox_test.go

```diff
@@ -20,116 +21,276 @@ import (
+		_, result := test.ShowForAuthClientClustersOK(t, svc.Context, svc, ctrl, clusterID)
+		// then
+		require.NotNil(t, result)
+		require.NotNil(t, result.Data)
+		assert.Equal(t, clusterPayload.Data.Name, result.Data.Name)
+		name := result.Data.Name
+		assert.Equal(t, httpsupport.AddTrailingSlashToURL(clusterPayload.Data.APIURL), result.Data.APIURL)
+		assert.Equal(t, httpsupport.AddTrailingSlashToURL(clusterPayload.Data.AppDNS), result.Data.AppDNS)
+		assert.Equal(t, false, result.Data.CapacityExhausted)
+		assert.Equal(t, fmt.Sprintf("https://console.cluster.%s/console/", name), result.Data.ConsoleURL)
+		assert.Equal(t, fmt.Sprintf("https://metrics.cluster.%s/", name), result.Data.MetricsURL)
+		assert.Equal(t, fmt.Sprintf("https://console.cluster.%s/console/", name), result.Data.LoggingURL)
+		assert.Equal(t, clusterPayload.Data.Type, result.Data.Type)
+		assert.Equal(t, clusterPayload.Data.AuthClientDefaultScope, result.Data.AuthClientDefaultScope)
+		assert.Equal(t, clusterPayload.Data.AuthClientID, result.Data.AuthClientID)
+		assert.Equal(t, clusterPayload.Data.AuthClientSecret, result.Data.AuthClientSecret)
+		assert.Equal(t, clusterPayload.Data.ServiceAccountToken, result.Data.ServiceAccountToken)
+		assert.Equal(t, clusterPayload.Data.ServiceAccountUsername, result.Data.ServiceAccountUsername)
+	})
+
+	s.T().Run("failure", func(t *testing.T) {
+
+		t.Run("not found", func(t *testing.T) {
+			// given
+			sa := &authtestsupport.Identity{
+				Username: authsupport.Auth,
+				ID:       uuid.NewV4(),
+			}
+			svc, ctrl := s.newSecuredControllerWithServiceAccount(sa)
+			// when/then
+			test.ShowForAuthClientClustersNotFound(t, svc.Context, svc, ctrl, uuid.NewV4())
+		})
+
+		t.Run("not allowed", func(t *testing.T) {
+			// given
+			sa := &authtestsupport.Identity{
+				Username: auth.Tenant,
+				ID:       uuid.NewV4(),
+			}
+			svc, ctrl := s.newSecuredControllerWithServiceAccount(sa)
+			// when/then
+			test.ShowForAuthClientClustersUnauthorized(t, svc.Context, svc, ctrl, clusterID)
+		})
+	})
+
 }
 
-func (s *ClusterControllerTestSuite) TestShowForAuthServiceAccountsOK() {
+func (s *ClustersControllerTestSuite) TestList() {
 	require.NotEmpty(s.T(), s.Configuration.GetClusters())
-	s.checkShowAuthForServiceAccount("fabric8-auth")
+
+	s.T().Run("ok", func(t *testing.T) {
+		for _, saName := range []string{"fabric8-oso-proxy", "fabric8-tenant", "fabric8-jenkins-idler", "fabric8-jenkins-proxy", "fabric8-auth"} {
+			t.Run(saName, func(t *testing.T) {
+				// given
+				sa := &authtestsupport.Identity{
+					Username: saName,
+					ID:       uuid.NewV4(),
+				}
+				svc, ctrl := s.newSecuredControllerWithServiceAccount(sa)
+				// when
+				_, clusters := test.ListClustersOK(t, svc.Context, svc, ctrl)
+				// then
+				require.NotNil(t, clusters)
+				require.NotNil(t, clusters.Data)
+				require.Len(t, clusters.Data, len(s.Configuration.GetClusters()))
+				for _, cluster := range clusters.Data {
```

Comment:
Also add at least on cluster to the DB manually and make sure it's among the result. Not only the clusters loaded from Configuration.

Reply:
ok, let me add that

Reply:
done in dc31c0f